code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -161,7 +161,8 @@
 
 	var/hit = 1
 	if(user.a_intent == I_HURT)
-		hit = ..()
+		. = ..()
+		hit = -1
```

Comment:
What does this being -1 do exactly?

Below, -1 is what is used for when the baton is off. Wouldn't this cause potential problems?

Reply:
I _think_ it is used to do a check for miss later on.
There is no check for miss on help intent.
~This code is barely readable, I did this quickly and it needs moar testing~